[BUG][TESTING]: test_clear_filters_functionality fails due to HTMX race condition

## Description

`tests/playwright/test_mcp_registry_page.py::TestMCPRegistryPage::test_clear_filters_functionality` fails intermittently with:

```
AssertionError: Clearing filters should show more or equal servers
assert 2 >= 28
```

After clearing filters, `get_server_count()` returns 2 instead of restoring to the initial (unfiltered) count.

## Root Cause

`MCPRegistryPage.clear_filters()` in `tests/playwright/pages/mcp_registry_page.py:237` resets all three filter controls without waiting for HTMX swaps between operations:

```python
def clear_filters(self) -> None:
    self.category_filter.select_option("")   # triggers HTMX swap
    self.auth_filter.select_option("")       # triggers another HTMX swap (races with first)
    self.search_input.fill("")
    self.page.wait_for_timeout(1000)
```

Each `select_option("")` triggers an HTMX request that replaces the `#server-grid` DOM. The second operation fires before the first swap completes, causing a race condition that leaves the grid in an incomplete state.

By contrast, the individual filter methods (`select_category`, `select_auth_type`) correctly wait for the swap:

```python
def select_category(self, category: str) -> None:
    self.category_filter.select_option(category)
    self.page.wait_for_selector("#server-grid", state="attached", timeout=30000)
```

## Proposed Fix

Add `wait_for_selector("#server-grid", state="attached", timeout=30000)` after each filter reset in `clear_filters()`, matching the pattern used by `select_category()` and `select_auth_type()`.

## Affected Files

- `tests/playwright/pages/mcp_registry_page.py` — `MCPRegistryPage.clear_filters()` method (line 237)
- `tests/playwright/test_mcp_registry_page.py` — `test_clear_filters_functionality` (line 168)

--- a/tests/playwright/pages/mcp_registry_page.py
+++ b/tests/playwright/pages/mcp_registry_page.py
@@ -242,6 +242,25 @@ class MCPRegistryPage(BasePage):
         self.page.wait_for_selector("#server-grid", state="attached", timeout=30000)
         self.search_input.fill("")
         self.page.wait_for_selector("#server-grid", state="attached", timeout=30000)
+
+        # Wait until the server grid count stabilizes (no change between two polls).
+        # This guards against HTMX partials that may reattach the grid multiple
+        # times and avoids races where the DOM is attached but not fully updated.
+        self.page.wait_for_function(
+            """() => {
+                const grid = document.querySelector('#server-grid');
+                if (!grid) return false;
+                const cnt = grid.querySelectorAll('.server-card').length;
+                if (window.__last_server_count_for_tests === cnt) {
+                    window.__last_server_count_for_tests = undefined;
+                    return true;
+                }
+                window.__last_server_count_for_tests = cnt;
+                return false;
+            }""",
+            timeout=30000,
+        )
+
         self.page.wait_for_timeout(1000)
 
     def click_category_badge(self, category: str) -> None:

